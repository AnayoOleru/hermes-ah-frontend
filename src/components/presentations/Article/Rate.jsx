/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/jsx-wrap-multilines */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';
import { Rating, Button, Popup } from 'semantic-ui-react';
import PropTypes from 'prop-types';

class Rate extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  rateHandler = (e, { rating }) => {
    // eslint-disable-next-line react/no-unused-state
    this.setState({ rating });
    this.rate(rating);
  };

  rate = rating => {
    const { articleId, rateArticle } = this.props;

    const data = {
      article_id: articleId,
      rating_value: rating,
    };

    rateArticle(data);
  };

  render() {
    const { likes } = this.props;

    return (
      <div className="rate-article">
        <div className="rate">
          <Rating maxRating={5} onRate={this.rateHandler} size="large" />
        </div>
        <div className="comment">
          <p>
            <i className="far fa-comment" />
          </p>
          <p>{likes}</p>
        </div>
        <Popup
          content="Report This Article"
          trigger={
            <div className="popout">
              <label htmlFor="popup">
                <i className="fas fa-ellipsis-v" />
              </label>
            </div>
          }
        />
        <input type="checkbox" id="popup" className="popout-control" />
        <div className="report-article">
          <Button>Report Article</Button>
        </div>
      </div>
    );
  }
}

Rate.defaultProps = {
  likes: 0,
};

Rate.propTypes = {
  articleId: PropTypes.string.isRequired,
  rateArticle: PropTypes.func.isRequired,
  likes: PropTypes.number,
};

export default Rate;
