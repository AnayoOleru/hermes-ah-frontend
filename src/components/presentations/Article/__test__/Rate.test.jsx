import React from 'react';
import { shallow } from 'enzyme';
import Rate from '../Rate';
import mock from '../../../../utils/rateDummyData';

describe('<Rate />', () => {
  it('should render rate component', () => {
    const wrapper = shallow(
      <Rate
        articleId={mock.articleId}
        rateArticle={mock.rateArticle}
        likes={mock.likes}
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });

  it('test the constructor', () => {
    const expectedProps = {
      articleId: 'asdf',
      rateArticle: () => {},
      likes: 1,
    };
    const rate = new Rate(expectedProps);
    expect(rate.state).toStrictEqual({});
  });

  it('describe rate function', () => {
    const wrapper = shallow(
      <Rate
        articleId={mock.articleId}
        rateArticle={mock.rateArticle}
        likes={mock.likes}
      />
    );
    expect(wrapper.instance().rate());
  });

  it('describe rateHandler function', () => {
    const wrapper = shallow(
      <Rate
        articleId={mock.articleId}
        rateArticle={mock.rateArticle}
        likes={mock.likes}
      />
    );
    const event = { preventDefault: jest.fn() };
    const value = { rating: 2 };
    expect(wrapper.instance().rateHandler(event, value));
  });
});
