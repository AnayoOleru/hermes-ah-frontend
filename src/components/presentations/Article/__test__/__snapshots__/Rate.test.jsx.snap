// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rate /> should render rate component 1`] = `ShallowWrapper {}`;
